File: src/js/request.ts
import promiseRetry = require('promise-retry');
import _request = require('request');

const domain = process.env["1337_domain"] || "https://1337x.to";

export function request(url): Promise<Response> {
    //const profiler = startProfiler();
    return promiseRetry((retry, num) => {
        return new Promise((resolve, reject) => {
            _request(domain + url, (err, response, html) => {
                if (err) {
                    reject(err);
                    return;
                }
                //profiler.done({message: 'server respond', url, statusCode: response.statusCode, try: num});
                resolve({html: html, statusCode: response.statusCode, try: num});
            });

        })
            .catch(retry);
    });
}

File: src/Domain/Request.fs
module Domain.Request

open Fable.Core

//add multi domain
type Url = Url of string

let unwrapUrl (Url a) = a

type Response =
    { html: string
      statusCode: int
      ``try``: int }

type Request = Url -> JS.Promise<Response>

module Url =
  let create (s: string) =
       Url <|
       if s.StartsWith "/" then
         s
       else
         sprintf "/%s" s


